src/chain/chain.rs

```diff
@@ -246,14 +246,16 @@ impl Chain {
             return Ok(Some(PollAccumulated::AccumulatedEvent(event)));
         }
 
-        if let Some(details) = self.poll_relocation() {
-            return Ok(Some(PollAccumulated::RelocateDetails(details)));
-        }
-
+        // Note: it's important that `promote_and_demote_elders` happens before `poll_relocation`,
+        // otherwise we might relocate a node that we still need.
```

Comment:
yes, that actually make more sense.

Reply:
Should I also make this order explicit somehow?

Reply:
The comment seem to be explicit enough to me?

Reply:
I'm thinking we are making the order explicit (statically enforced) here: https://github.com/maidsafe/routing/pull/1960#discussion_r359428920, so we might as well do it here too. Or do you think it's not worth it?

Reply:
The other one had an actual data dependency that was implicit, this one does not.
I'm not sure that would be worth trying to add something here. (Also the other was across multiple functions)